### PR TITLE
Fix: Scope raises error when array is passed as argument.

### LINKED ISSUE #25654: Scope raises error when array is passed as argument
### Steps to reproduce

``` ruby
begin
  require 'bundler/inline'
rescue LoadError => e
  $stderr.puts 'Bundler version 1.10 or later is required. Please update your Bundler'
  raise e
end

gemfile(true) do
  source 'https://rubygems.org'
  gem 'activerecord', '5.0.0'
  gem 'sqlite3'
end

require 'active_record'
require 'minitest/autorun'
require 'logger'

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: 'sqlite3', database: ':memory:')
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
  end

  create_table :comments, force: true do |t|
    t.integer :post_id
  end
end

class Post < ActiveRecord::Base
  has_many :comments
end

class Comment < ActiveRecord::Base
  belongs_to :post

  scope :by_post_ids, -> (post_ids = []) { where(post_id: post_ids) }
end

class BugTest < Minitest::Test
  def test_array_as_argument_to_scope
    Comment.by_post_ids([1, 2, 3])
  end
end
```
### Expected behavior

It returns comments for `post_ids` passed as argument
### Actual behavior

Raise 

```
ArgumentError: wrong number of arguments (3 for 0..1)
```
### System configuration

**Rails version**: 5.0.0

**Ruby version**: 2.2.4

### PR DESCRIPTION
**Pull request by mechanicles**
Fixes #25654

~~Pass scope's arguments as it is to the proc body.~~

Update: Don't splat the args if argument size is 1 and that argument is an array.

### CODE REVIEW

**Comment by maclover7** [comment]
Looks like this broke some Active Record tests. Would you mind taking a look at the build failures on Travis CI? 😬

**Comment by mechanicles** [comment]
@maclover7 Yes I will :)

**Comment by mechanicles** [comment]
@maclover7 Fixed the build failures on the Travis CI.

File: activerecord/test/models/comment.rb [hunk 1]
```diff
@@ -19,6 +19,8 @@ class Comment < ActiveRecord::Base
   has_many :children, :class_name => 'Comment', :foreign_key => :parent_id
   belongs_to :parent, :class_name => 'Comment', :counter_cache => :children_count
 
+  scope :by_post_ids, -> (post_ids = []) { where(post_id: post_ids) }
```
Comment (kamipo) [inline activerecord/test/models/comment.rb]: Probably this is a bug `lambda` with `instance_exec`. It works using `proc`.

```ruby
  scope :by_post_ids, proc { |post_ids = []| where(post_id: post_ids) }
```

**Comment by nurse** [comment]
This can be simplified as `instance_exec([1,2,3], &->(a=[]){ p a })`.
It correctly works on Ruby 2.1 but raises ArgumentError on Ruby 2.2, 2.3 and trunk (2.4).
I fixed it at https://github.com/ruby/ruby/commit/c016898e79edec4fe3c5106ba7e7742a5eaa3f17

**Comment by mechanicles** [comment]
@sgrif @tenderlove Any update on this? because actual issue got fixed in Ruby a few days back. 

Is this PR still valid? We also need to support for old/current Ruby versions as well so I think this PR makes sense to me.

Let me know your thoughts.

**Comment by krzysiek1507** [comment]
What about this?

**Comment by nurse** [comment]
After Ruby 2.2.6/2.3.2 is released, this PR will be obsoleted.
But they are not released yet.

**Comment by arthurnn** [comment]
@tenderlove should we merge this anyways with a comment to remove once ruby 2.3.2 is out?

**Comment by kamipo** [comment]
Ruby 2.3.2 has been released!
https://www.ruby-lang.org/en/news/2016/11/15/ruby-2-3-2-released/

**Pull request closed without merging**